llvm/lib/Transforms/Vectorize/VectorCombine.cpp

```diff
@@ -1155,9 +1161,6 @@ bool VectorCombine::scalarizeLoadExtract(Instruction &I) {
     if (!UI || UI->getParent() != LI->getParent())
       return false;
 
-    if (!isGuaranteedNotToBePoison(UI->getOperand(1), &AC, LI, &DT))
```

Comment:
This check will prevent the transform, but will be checked later inside `canScalarizeAccess`, so it is safe to remove.